robotmbt/visualise/networkvisualiser.py

```diff
@@ -1,450 +1,460 @@
-    EXECUTED_LABEL_COLOR = 'black'
-    UNEXECUTED_LABEL_COLOR = '#A9A9A9'
-
-    def __init__(self, graph: AbstractGraph, suite_name: str = ""):
-        self.plot = None
-        self.graph = graph
-        self.suite_name = suite_name
-        self.node_props = {}  # Store node properties for arrow calculations
-        self.graph_layout = {}
-
-        # graph customisation options
-        self.node_radius = 1.0
-        self.char_width = 0.1
-        self.char_height = 0.1
-        self.padding = 0.1
-
-        # Get executed elements for visual differentiation
-        final_trace = graph.get_final_trace()
-        self.executed_nodes = set(final_trace)
-        self.executed_edges = set()
-        for i in range(0, len(final_trace) - 1):
-            from_node = final_trace[i]
-            to_node = final_trace[i + 1]
-            self.executed_edges.add((from_node, to_node))
-
-    def generate_html(self) -> str:
-        """
-        Generate html file from networkx graph via Bokeh
-        """
-        self._calculate_graph_layout()
-        self._initialise_plot()
-        self._add_nodes_with_labels()
-        self._add_edges()
-        return file_html(self.plot, CDN, "graph")
-
-    def _initialise_plot(self):
-        """
-        Define plot with width, height, x_range, y_range and enable tools.
-        x_range and y_range are padded. Plot needs to be a square
-        """
-        padding: float = self.GRAPH_PADDING_PERC / 100
-
-        x_range, y_range = zip(*self.graph_layout.values())
-        x_min = min(x_range) - padding * (max(x_range) - min(x_range))
-        x_max = max(x_range) + padding * (max(x_range) - min(x_range))
-        y_min = min(y_range) - padding * (max(y_range) - min(y_range))
-        y_max = max(y_range) + padding * (max(y_range) - min(y_range))
-
-        # scale node radius based on range
-        nodes_range = max(x_max - x_min, y_max - y_min)
-        self.node_radius = nodes_range / 150
-        self.char_width = nodes_range / 150
-        self.char_height = nodes_range / 150
-
-        # create plot
-        x_range = Range1d(min(x_min, y_min), max(x_max, y_max))
-        y_range = Range1d(min(x_min, y_min), max(x_max, y_max))
-
-        self.plot = Plot(width=self.GRAPH_SIZE_PX,
-                         height=self.GRAPH_SIZE_PX,
-                         x_range=x_range,
-                         y_range=y_range)
+from grandalf.graphs import Vertex as GVertex, Edge as GEdge, Graph as GGraph
+from grandalf.layouts import SugiyamaLayout
+
+from networkx import DiGraph
+
+from robotmbt.visualise.graphs.abstractgraph import AbstractGraph
+
+# Padding within the nodes between the borders and inner text
+HORIZONTAL_PADDING_WITHIN_NODES = 5
+VERTICAL_PADDING_WITHIN_NODES = 5
+
+# Colors for different parts of the graph
+FINAL_TRACE_NODE_COLOR = '#CCCC00'
+OTHER_NODE_COLOR = '#999989'
+
+# Dimensions of the plot in the window
+INNER_WINDOW_WIDTH = 846
+INNER_WINDOW_HEIGHT = 882
```

Comment:
why don't we use these to caculate the width/height of the plot in `Visualiser.py`?